client/components/banner.jsx

```diff
@@ -57,38 +29,36 @@ export default class Banner extends React.Component {
         }
       }
     };
-    this.getTrailInfo = this.getTrailInfo.bind(this);
   }
 
   componentDidMount() {
-    // this.getTrailInfo();
-  }
-
-  getTrailInfo() {
     fetch('http://localhost:3001/1/trailInfo')
       .then((response) => {
-        response.json()
-          .then((res) => {
-            this.setState((prevState) => ({
-              trail: res.data
-            }));
-            // console.log('this.state.trail: ', res.data);
-          });
+        return response.json();
+      })
+      .then((json) => {
+        this.setState( (prevState) => ({
+          trail: json.data,
+          fetching: false
+        }));
+        // console.log('json :', json.data);
+      })
+      .catch((error) => {
+        console.error('getTrailInfo error: ', error);
       });
   }
 
   render() {
-    return (
+    return (this.state.fetching) ? (<div className="fetching"></div>) : (
       <div>
         {/* Template strings (template literals) used for css module format of Bootstrap class names */}
         {/* See webpack.config.js css-loader options for localIdentName syntax of css output */}
-        {/* <!-- text/overlay content on banner --> */}
 
         <div id="banner_content" className="row flex-row justify-content-left align-items-end col-12">
           <div className="jumbotron-fluid d-flex">
             <img className="heroPhoto img-fluid" src={this.state.heroPhoto.attributes.photoUrl} alt="hero img" />
```

Comment:
it might be easier to make your banner image as a div, with a css style `background: url('')`. but you can do it this way too.

Reply:
Good point - initially, I did a bit of research on that, and it seems the most common decision point comes down to _"is it content, or just stylistic?"_ imho, I consider the hero image content (clickable in the original source implementation, and a part of the "detail page" for a trail) so I went with the inline `img` element. Would be interesting to revisit this again during optimization phase - no idea if there's any significant impact to performance based on background or inline.